cmd/root.go

```diff
@@ -78,23 +76,37 @@ func SetLogLevel(logLevel string) {
 
 // initConfig reads in config file and ENV variables if set.
 func initConfig() {
-	if cfgFile != "" {
-		// Use config file from the flag.
-		viper.SetConfigFile(cfgFile)
-	} else {
-		// Find home directory.
-		home, err := homedir.Dir()
-		cobra.CheckErr(err)
-
-		// Search config in home directory with name ".swinch" (without extension).
-		viper.AddConfigPath(home)
-		viper.SetConfigName(".swinch")
-	}
+	// Search config in "~/.swinch/config.yaml".
+	viper.AddConfigPath(config.HomeFolder() + config.CfgFolderName)
+	viper.SetConfigName("config.yaml")
+	viper.SetConfigType("yaml")
 
 	viper.AutomaticEnv() // read in environment variables that match
 
-	// If a config file is found, read it in.
-	if err := viper.ReadInConfig(); err == nil {
-		fmt.Fprintln(os.Stderr, "Using config file:", viper.ConfigFileUsed())
+	ds := domain.Datastore{}
+
+	switch ds.FileExists(config.HomeFolder() + config.CfgFolderName + config.CfgFileName) {
+	case true:
+		// If a config file is found, read it in and validate current-context against available contexts
+		if err := viper.ReadInConfig(); err == nil {
```

Comment:
This nested if statement can be simplified and written as two independent if statements - ie make the code simpler.
ex:  
`err := viper.ReadInConfig() ; 
if err!= nill {log.Fatalf("A parsing error"}
contextExists := config.ValidateCurrentContext()
if contextExists != true {log.Fatalf}
log.Debugf()
`